odata_client_tracing/src/main/java/com/sdl/odata/client/caller/TracingEndpointCaller.java

```diff
@@ -0,0 +1,226 @@
+import com.github.kristofa.brave.httpclient.BraveHttpResponseInterceptor;
+import com.sdl.odata.api.service.HeaderNames;
+import com.sdl.odata.api.service.MediaType;
+import com.sdl.odata.api.service.ODataRequest;
+import com.sdl.odata.client.api.caller.EndpointCaller;
+import com.sdl.odata.client.api.exception.ODataClientException;
+import com.sdl.odata.client.api.exception.ODataClientRuntimeException;
+import org.apache.http.HttpHost;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.sdl.odata.api.service.MediaType.ATOM_XML;
+import static com.sdl.odata.api.service.MediaType.XML;
+import static com.sdl.odata.client.ODataClientConstants.DefaultValues.CLIENT_PROXY_PORT_DEFAULT;
+import static com.sdl.odata.client.ODataClientConstants.DefaultValues.CLIENT_TIMEOUT_DEFAULT;
+import static com.sdl.odata.client.ODataClientConstants.WebService.CLIENT_CONNECTION_TIMEOUT;
+import static com.sdl.odata.client.ODataClientConstants.WebService.CLIENT_SERVICE_PROXY_HOST_NAME;
+import static com.sdl.odata.client.ODataClientConstants.WebService.CLIENT_SERVICE_PROXY_PORT;
+import static com.sdl.odata.client.property.PropertyUtils.getIntegerProperty;
+import static com.sdl.odata.client.property.PropertyUtils.getStringProperty;
+import static com.sdl.odata.client.util.ODataClientUtils.buildException;
+import static com.sdl.odata.client.util.ODataClientUtils.closeIfNecessary;
+import static com.sdl.odata.client.util.ODataClientUtils.populateRequestProperties;
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static org.apache.http.util.TextUtils.isBlank;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Tracing implementation of {@link com.sdl.odata.client.api.caller.EndpointCaller}.
+ * Using brave to generate and send spans into zipkin.
+ */
+public class TracingEndpointCaller implements EndpointCaller {
+
+    private static final Logger LOG = getLogger(TracingEndpointCaller.class);
+
+    private static final String WRONG_URL_MESSAGE = "The URL syntax is wrong";
+    private static final String REQUEST_FAILED_MESSAGE = "Cannot make a request to URL: ";
+    private static final String APPLICATION_PROPERTIES_FILE_NAME = "/config/application.properties";
+    private static final String DEFAULT_ZIPKIN_HOSTNAME = "http://192.168.99.100:9411";
```

Comment:
Well for zipkin the default web app port number is 9411. And 192.168.99.100 stands for docker host address by default. Do you want to delete it at all or use localhost?